src/map/script.cpp

```diff
@@ -2737,9 +2737,12 @@ struct script_data *get_val_(struct script_state* st, struct script_data* data,
 			case '\'':
 				{
 					unsigned short instance_id = script_instancegetid(st);
-					if( instance_id )
-						data->u.str = (char*)i64db_get(instance_data[instance_id].regs.vars,reference_getuid(data));
-					else {
+
+					if (instance_id) {
+						auto idata = instance_search(instance_id);
```

Comment:
`instance_search` is guaranteed that the return pointer is a valid one? Ditto elsewhere